### DISCUSSION TITLE
OCR works but not actually enabled

### DISCUSSION THREAD

**Discussion by igiuseppem** [sepinf-inc/IPED#896] [General]
Good morning.

I have bulilded successfully the latest version of IPED. I have indexed a forensic image. After the process, I found that I can search italian terms inside the files. It means that IPED executed the OCR, but in the conf files of this build I haven't activated this option and I don't want it executes on all case. What can I check to understand this behavior?
In the previous version of IPED I have activated the OCR option setting italian language. Maybe it's something linked to this? I don't figure how it is possible, since the directory that contains the new build it's different from the previous and in the conf file I have "enableOCR = false".

Thank you very much
Happy new year!

**Comment by wladimirleite** [comment]
I guess the first thing to check is the processing log, and look for OCR. It should tell if it was really enabled.

What is the type of file that you believe is being OCRed? Can you share one of these files?

**Comment by igiuseppem** [comment]
In the log (found in \IPED\target\release\iped-4.0.0-snapshot\log) the only reference to ocr is this warning "[WARN]	[indexer.parsers.PDFOCRTextParser]			Plugin JPEG2000 not found, JPX images will not be decoded from PDFs. You can download it from https://mvnrepository.com/artifact/com.github.jai-imageio/jai-imageio-jpeg2000/1.3.0 and put it in plugins folder. Warn: that plugin is worse to decode JPX outside of PDFs!"
About the file that are OCRed, they are pdf, txt, xls, doc, html. I have searched various terms and I found several findings. Sorry, I can't share the files.

**Reply by lfcnassif** [reply]
> the only reference to ocr is this warning...

After the refactoring on #555 some options stopped to be logged. I'll open an issue to log them and other configs.

> pdf, txt, xls, doc, html

Txt and html are never submitted to OCR. This also applies to xls/doc without images. They are indexed after default text extraction, OCR is not needed to index text files. To know if OCR was run or not, check the iped/text/ocr-results.db sqlite database and see if it is populated.

**Comment by igiuseppem** [comment]
Thank you all.
The ocr-results.db doesn't exist, so I guess the OCR was not running. Nevertheless, if I search for terms in the forensic copy indexed with IPED, I obtains also results from content of pdf files.

**Reply by lfcnassif** [reply]
Common PDFs with text data also don't need OCR to be indexed. As ocr-results.db doesn't exist, probably OCR wasn't executed.

**Reply by igiuseppem** [reply]
Ok, thank you. Effectively they are pdf with text data.
